### PR TITLE
Monitor permissions

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
To aid in debugging deployments, let's periodically check the application permissions and log those missing or redundant.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-59549

### CODE REVIEW

File: server/credentials.go [hunk 1]
```diff
@@ -1,14 +1,95 @@
...
+	case ResourceAccessTypeScope:
+		return "Scope (Delegated)"
+	default:
+		return resourceAccess.Type
+	}
+}
+
+// getExpectedPermissions returns the set of expected permissions, keyed by the
+// name the enduser would expect to see in the Azure tenant.
+func getExpectedPermissions() []expectedPermission {
```
Comment (lieut-data) [inline server/credentials.go:46]: This is hard-coded for now, but I anticipate the permission set changing as we add functionality, some of which may be optional.

**Review by enahum — APPROVED** [review]
Lgtm but I think as is related to some permissions and logging perhaps it makes sense to include a security review?

**Comment by esarafianou** [comment]
@lieut-data yes that works considering we'll start the review in the beginning of August

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Pull request merged**